api/src/context.rs

```diff
@@ -1153,7 +1153,7 @@ impl Context {
                 .map_err(|e| {
                     E::internal_with_code(e, AptosErrorCode::InternalError, ledger_info)
                 })?;
-            let storage_adapter = StorageAdapter::new(&state_view);
+            let storage_adapter: StorageAdapter<_, ()> = StorageAdapter::new(&state_view);
```

Comment:
you can use `state_view.as_move_resolver()` here (I think that was the intention of that trait, to hide the adapter, but not sure)